File: javascript/quicksort.js
function particion(arr, bajo, alto) {
    let pivote = arr[alto];
    let i = bajo - 1;

    for (let j = bajo; j < alto; j++) {
        if (arr[j] < pivote) {
            i++;
            [arr[i], arr[j]] = [arr[j], arr[i]];
        }
    }
    [arr[i + 1], arr[alto]] = [arr[alto], arr[i + 1]];
    return i + 1;
}

function quickSort(arr, bajo = 0, alto = arr.length - 1) {
    if (bajo < alto) {
        let pi = particion(arr, bajo, alto);
        quickSort(arr, bajo, pi - 1);
        quickSort(arr, pi + 1, alto);
    }
    return arr;
}

const arr = [10, 7, 8, 9, 1, 5];
console.log(quickSort(arr));

File: c++/quicksort.cpp
#include <iostream>
#include <vector>

int particion(std::vector<int>& arr, int bajo, int alto) {
    int pivote = arr[alto];
    int i = bajo - 1;

    for (int j = bajo; j < alto; j++) {
        if (arr[j] < pivote) {
            i++;
            std::swap(arr[i], arr[j]);
        }
    }
    std::swap(arr[i + 1], arr[alto]);
    return i + 1;
}

void quickSort(std::vector<int>& arr, int bajo, int alto) {
    if (bajo < alto) {
        int pi = particion(arr, bajo, alto);
        quickSort(arr, bajo, pi - 1);
        quickSort(arr, pi + 1, alto);
    }
}

int main() {
    std::vector<int> arr = {10, 7, 8, 9, 1, 5};
    quickSort(arr, 0, arr.size() - 1);
    for (int x : arr) std::cout << x << " ";
    return 0;
}